app/repositories/SessionRepository.scala

```diff
@@ -16,73 +16,77 @@
-  implicit val formats: OFormat[DatedCacheMap] = Json.format[DatedCacheMap]
+                         lastUpdated: LocalDateTime = LocalDateTime.now(ZoneId.of("UTC"))) extends MongoFormats {
 
-  def apply(cacheMap: CacheMap): DatedCacheMap = DatedCacheMap(cacheMap.id, cacheMap.data)
+  def toCacheMap: CacheMap = {
+    CacheMap(this.id, this.data)
+  }
 }
 
-class ReactiveMongoRepository(config: Configuration, mongo: () => DefaultDB)
-  extends ReactiveRepository[DatedCacheMap, BSONObjectID](config.get[String]("appName"), mongo, DatedCacheMap.formats) {
+object DatedCacheMap {
+  implicit def apply(cacheMap: CacheMap): DatedCacheMap = DatedCacheMap(cacheMap.id, cacheMap.data)
 
-  val fieldName = "lastUpdated"
-  val createdIndexName = "userAnswersExpiry"
-  val expireAfterSeconds = "expireAfterSeconds"
-  val timeToLiveInSeconds: Int = config.get[Int]("mongodb.timeToLiveInSeconds")
+  implicit val dateFormat: Format[LocalDateTime] = MongoJavatimeFormats.localDateTimeFormat
+  implicit val formats: OFormat[DatedCacheMap] = Json.format[DatedCacheMap]
+}
 
-  createIndex(fieldName, createdIndexName, timeToLiveInSeconds)
+class MongoRepository(config: Configuration, mongo: MongoComponent)
+  extends PlayMongoRepository[DatedCacheMap](
+    mongoComponent = mongo,
+    collectionName = config.get[String]("appName"),
+    domainFormat = DatedCacheMap.formats,
+    indexes = Seq(IndexModel(
+      ascending(config.get[String]("mongodb.fieldName")),
+      IndexOptions()
+        .name(config.get[String]("mongodb.indexName"))
+        .expireAfter(config.get[Int]("mongodb.timeToLiveInSeconds"), TimeUnit.SECONDS)
+    )),
+    replaceIndexes = false) {
 
-  private def createIndex(field: String, indexName: String, ttl: Int): Future[Boolean] = {
-    collection.indexesManager.ensure(Index(Seq((field, IndexType.Ascending)), Some(indexName),
-      options = BSONDocument(expireAfterSeconds -> ttl))) map {
-      result => {
-        logger.debug(s"set [$indexName] with value $ttl -> result : $result")
-        result
-      }
-    } recover {
-      case e => logger.error("Failed to set TTL index", e)
-        false
+  def upsert(cm: CacheMap): Future[Boolean] = {
+    val cmUpdated = DatedCacheMap(cm.id, cm.data)
+    val options = ReplaceOptions().upsert(true)
+    collection.replaceOne(equal("id", cm.id), cmUpdated, options).toFuture().map { result =>
+      result.wasAcknowledged()
     }
   }
 
-  def upsert(cm: CacheMap): Future[Boolean] = {
-    val selector = BSONDocument("id" -> cm.id)
-    val cmDocument = Json.toJson(DatedCacheMap(cm))
-    val modifier = BSONDocument("$set" -> cmDocument)
-    collection.update(ordered=false).one(selector, modifier, upsert = true).map { lastError =>
-      lastError.ok
+  def get(id: String): Future[Option[CacheMap]] = {
+    collection.find(equal("id", id)).headOption().map { datedCacheMap =>
+      Try(datedCacheMap.get.toCacheMap) match {
```

Comment:
Could you just map over datedCacheMap here rather than calling .get?

Reply:
Definitely could yep, I think I was trying to avoid nesting the maps but happy to swap it over if it's more readable that way